src/main/java/io/micrometer/newrelic/NewRelicRegistry.java

```diff
@@ -263,22 +263,30 @@ public NewRelicRegistry build() {
     }
 
     private MetricBatchSender createMetricBatchSender() {
-      MetricBatchSenderBuilder metricBatchSenderBuilder =
-          MetricBatchSender.builder()
+      SenderConfiguration.SenderConfigurationBuilder builder =
+          MetricBatchSender.configurationBuilder()
               .apiKey(config.apiKey())
               .httpPoster(new MicrometerHttpPoster(httpSender))
-              .secondaryUserAgent("NewRelic-Micrometer-Exporter", implementationVersion);
-      if (config.enableAuditMode()) {
-        metricBatchSenderBuilder.enableAuditLogging();
+              .secondaryUserAgent("NewRelic-Micrometer-Exporter/" + implementationVersion)
+              .auditLoggingEnabled(config.enableAuditMode());
+      builder = configureEndpoint(builder);
+      return MetricBatchSender.create(builder.build());
+    }
+
+    private SenderConfiguration.SenderConfigurationBuilder configureEndpoint(
+        SenderConfiguration.SenderConfigurationBuilder builder) {
+      if (config.uri() == null) {
+        return builder;
       }
-      if (config.uri() != null) {
-        try {
-          metricBatchSenderBuilder.uriOverride(URI.create(config.uri()));
-        } catch (MalformedURLException e) {
-          throw new RuntimeException("Invalid URI for the metric API : " + config.uri(), e);
+      try {
+        URI uri = URI.create(config.uri());
+        if (uri.getPath() != null && uri.getPath().length() > 0) {
+          return builder.endpointWithPath(uri.toURL());
         }
+        return builder.endpoint(uri.getScheme(), uri.getHost(), uri.getPort());
+      } catch (MalformedURLException e) {
+        throw new RuntimeException("Invalid URI for the metric API : " + config.uri(), e);
       }
```

Comment:
Given I've seen these lines of code twice in 2 days, in two different projects, should we consider making this hunk of logic part of the SDK configuration API directly?

Reply:
Absolutely!  My thoughts exactly.

Reply:
I'll put an issue in for that.